### ISSUE TITLE
Component Inheritance

### ISSUE THREAD

**Issue by scttcper** [scttcper/ngx-toastr#39]
Figure out what that is in v2.3.0 and demo how to create custom toasts

**Comment by cmckni3** [comment]
This is the neat thing about this project vs ng2-toastr.

My only complaint is component inheritance requires a bit of effort because of the component decorator.

**Comment by scttcper** [comment]
I've cleaned up the example for 2.3. 
https://github.com/scttcper/toastr-ng2/blob/master/src/app/pink.toast.ts

No more constructor or component hosts.

Annoying to copy the template and animations.

**Comment by cmckni3** [comment]
This might be worth revisiting since the latest version uses DI to inject more dependencies than previously. Some of the dependencies are actually private (`ToastData`, `ToastRef`).

**Comment by cmckni3** [comment]
For my use case, I think composition would be better. I added a custom button and click handler to the template.

**Comment by scttcper** [comment]
Think we need a button that isn't close. Its not in the original toastr, but i see it in every other library's toast. Like blueprint http://blueprintjs.com/docs/#components.toaster

<img width="313" alt="screen shot 2017-02-20 at 7 41 57 pm" src="https://cloud.githubusercontent.com/assets/1400464/23149993/b6e6880a-f7a4-11e6-86dc-db39e3c07614.png">

**Comment by yarrgh** [comment]
I like the idea of having the extra button. Could definitely find use cases for it.

**Comment by scttcper** [comment]
Released 4.3.0 with onAction that can be used in custom toasts. onAction is passed back the same way onTap is except it isn't used anywhere in the default template. The pink toast in the demo shows how it might be used. You will probably want to combine this with `tapToDismiss = false` or capture the click event and stop propagation.

```html
<a (click)="action()">Action Name</a>
```
```ts
action() {
  this.onAction.next('helpful string');
  this.onAction.complete();
}
```

**Comment by WWind220** [comment]
Can you please help me with such usage: I am using custom toast with properties
`
 opt.extendedTimeOut = 0;
 opt.timeOut = 0;
 opt.tapToDismiss = false;

`
So, I want to have full control on the action of closing toast. How can I do it? Is there any function for this?
